persistent/Database/Persist/Types/Base.hs

```diff
@@ -377,6 +377,7 @@ data PersistValue = PersistText Text
                   | PersistMap [(Text, PersistValue)]
                   | PersistObjectId ByteString -- ^ Intended especially for MongoDB backend
                   | PersistArray [PersistValue] -- ^ Intended especially for PostgreSQL backend for text arrays
+                  | PersistLiteral ByteString -- ^ Using 'PersistLiteral' you can customize PersistField instances to output unescaped SQL
                   | PersistDbSpecific ByteString -- ^ Using 'PersistDbSpecific' allows you to use types specific to a particular backend
```

Comment:
Okay, here's a migration plan that can add this feature safely *and* manage #1123

1. `PersistLiteral` is introduced as *the* "unescaping" variant.
2. We add a constructor `PersistLiteralEscaped` that has the escaping behavior consistently across DB packages.
3. Add a deprecation notice to `PersistDbSpecific` specifying what's changed and how. Mention that it will be removed in the next major version release of the library. This warning should percolate through the ecosystem and packages will update.

Then, in the next major version, we can make the behavior breaking changes in the relevant database packages, unifying how postgres/mysql/sqlite use these constructors.

Reply:
Will add.